---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

# curatedMetagenomicDataTerminal

<!-- badges: start -->
<!-- badges: end -->

The curatedMetagenomicData command-line interface provides standardized, curated human microbiome data for novel analyses. It includes gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples collected from different body sites. The bacterial, fungal, and archaeal taxonomic abundances for each sample were calculated with MetaPhlAn3, and metabolic functional potential was calculated with HUMAnN3. The manually curated sample metadata and standardized metagenomic data are available through this command-line interface.

## Prerequisite

R is required; visit https://cran.r-project.org/ for installation instructions.

## Installation

The curatedMetagenomicData command-line interface, curatedMetagenomicDataTerminal, is simply an R package with a symlinked R script that uses the curatedMetagenomicData R package to return data for use in other languages (python, etc.). You can install it the easier way (one command) or the harder way (three commands); neither way is difficult.

### Easier Way

The easier way pipes the results of `curl` to `bash`, and that is always met with some controversy. If you don't like it, don't do it; the harder way is right there for you and is identical to the easier way, only it doesn't check for R first. Follow the link to the install script (https://tinyurl.com/cMDTerminal) if you need to prove this to yourself before executing the one line install.

```{sh, eval = FALSE}
curl -sSL https://tinyurl.com/cMDTerminal | bash
```

### Harder Way

As stated, the harder way is identical to the easier way, it just assumes you already have R installed. To install curatedMetagenomicDataTerminal the harder way, execute the following three commands from your terminal.

```{sh, eval = FALSE}
Rscript -e 'utils::install.packages("BiocManager", repos = "https://cloud.r-project.org/")'
Rscript -e 'BiocManager::install("waldronlab/curatedMetagenomicDataTerminal")'
Rscript -e 'curatedMetagenomicDataTerminal::install()'
```
### Windows Installation

Installation on Windows is possible but requires a bit of work:

1. Add R to your Path.
    + Press the Windows key and type "path" then select "Edit the system environment variables."
    + Under Advanced, select "Environment Variables..." toward the bottom.
    + The lower half of the screen should be "System variables." Select Path and then select "Edit..."
    + Select New then enter the path to your R installation bin folder (typically C:\Program Files\R\R-<Version>\bin).
    
2. Ensure that it was added to your Path by opening Command Prompt or Powershell and running `Rscript --version`.

3. Run these lines. Note that the position of the single and double quotes have been reversed from the same code block above. This will not work on Windows otherwise.
```{sh, eval = FALSE}
Rscript -e "utils::install.packages('BiocManager', repos = 'https://cloud.r-project.org/')"
Rscript -e "BiocManager::install('waldronlab/curatedMetagenomicDataTerminal')"
```

4. Create a symbolic link to the curatedMetagenomicData script to do this run the following line of code in a Powershell window with elevated privileges (to run Powershell with elevated privileges, right click and choose "Run as Administrator").
```{sh, eval = FALSE}
New-Item -ItemType SymbolicLink "curatedMetagenomicData" -Target "<path to R library>/curatedMetagenomicDataTerminal/exec/curatedMetagenomicData"
```

5. Check to make sure everything is setup correctly by running this line in Powershell:
```{sh, eval = FALSE}
Rscript curatedMetagenomicData --help
```


## Introduction

```{bash, comment = "", echo = FALSE}
./exec/curatedMetagenomicData --help
```

## Code of Conduct

Please note that the curatedMetagenomicDataTerminal package is released with a code of conduct. By contributing, you agree to abide by its terms.
